### PR TITLE
feat: add time travel queries

### PR DESCRIPTION
**Pull request by Sekhmet**
Closes: https://github.com/checkpoint-labs/checkpoint/issues/61

This PR adds support for time travel queries by storing mutations of entities with ranges those are valid for.
You can specify `block` filter` when querying API to get version at given block.

This will be also used for reorg support to be able to revert changes.

Changes:
- MySQL support is removed to use PostgreSQL's native `int8range` type.
- Direct `pg` access in writers is removed - without knowing the exact inner-workings of our data structure queries on SQL level will most likely result in different effect that user will expect. All queries should be done through ORM that encapsulates all voodoo we have to do.
- `bluebird` was removed (wanted to remove it earlier, but we used it to promisify mysql, now it was removed entirely and only usage was replaced by simple `sleep` helper. Now this gives us better type-safety.
- `block` filter is available for single and multi entity queries.

### Examples

Test with this branch which has more complex schema to test all edge-cases: https://github.com/checkpoint-labs/checkpoint-template/compare/sekhmet/tt?expand=1

```gql
{
  users (block: 65309) {
    id
    created_at
    created_at_block
    posts {
      id
      content
      created_at_block
      tags {
        id
      }
    }
    info {
      post_count
    }
  }
}
```

### CODE REVIEW

File: src/checkpoint.ts [hunk 1]
```diff
@@ -299,12 +296,11 @@ export default class Checkpoint {
   public async getWriterParams(): Promise<{
     instance: Checkpoint;
     knex: Knex;
-    mysql: AsyncMySqlPool;
     pg: PgPool;
   }> {
     return {
       instance: this,
-      mysql: this.mysql,
+      knex: this.knex,
```
Comment (Sekhmet) [inline src/checkpoint.ts:303]: We should remove knex and pg from there as well.

**Comment by bonustrack** [comment]
Did some small test and it seem to work well. I noticed using `block: 0` would return all the results, I don't think that should be the case but it's just a detail.

**Review by bonustrack — APPROVED** [review]
tACK, tested with the branch `sekhmet/tt` from Checkpoint template adding some entity that get updated and so far its working well

```gql
type Overview {
  id: String!
  post_count: Int!
}
```

**Comment by Sekhmet** [comment]
@bonustrack I updated checkpoint-template to use more complex scenario that should cover all cases (with sample query in this block post).

Also handled `0` block and added schema version verification so database will be reset once this version gets introduced automatically (because otherwise database will be in broken state).

If it all still looks good to you I will merge this PR, do release and migrate our APIs to this version.

**Comment by bonustrack** [comment]
Also does it support delete entity?

**Comment by Sekhmet** [comment]
> Also does it support delete entity?

Yes, you can delete entities.

> Looks good, does this version works for subqueries assuming range?

Not sure if I understand it, but `block` you specify affects all subentities (so for example in sample query when you request `User` with `UserInfo` at specific block that `UserInfo` will be version for the block you requested).
